### PR TITLE
Fix message details layout in iPhone X.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

**Comment by michaelkirk-signal** [comment]
LGTM

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m [hunk 1]
```diff
@@ -46,13 +46,7 @@ - (void)viewDidLoad
     self.tableView.rowHeight = UITableViewAutomaticDimension;
     self.tableView.estimatedRowHeight = 70;
 
-    // Fix a bug that only affects iOS 11.0.x and 11.1.x.
-    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(11, 0) && !SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(11, 2)) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wpartial-availability"
-        self.tableView.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentNever;
-#pragma clang diagnostic pop
-    }
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: > Just checking - did you check that we have the appropriate content inset behavior?

Are you referring to this fix?

Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: I've DRY'd it up in a category.

File: SignalMessaging/categories/UIView+OWS.m [hunk 2]
```diff
@@ -487,4 +487,21 @@ - (void)logHierarchyUpwardLaterWithLabel:(NSString *)label
 
 @end
 
+#pragma mark -
+
+@implementation UIScrollView (OWS)
+
+- (void)applyScrollViewInsetsFix
+{
+    // Fix a bug that only affects iOS 11.0.x and 11.1.x.
```
Comment (charlesmchen-signal) [inline SignalMessaging/categories/UIView+OWS.m]: Am I right in understanding that we should use this for any scroll view that:

* Could be the bottom-most scroll view of a given view.
* Isn't the root view for its view controller?

In my testing, it looks like we also want to apply to this 11.2.  If we don't, we end up with too much content inset.  Let's discuss?

Comment (michaelkirk-signal) [reply SignalMessaging/categories/UIView+OWS.m]: This fix, as I recall, was not to fix the *amount* of content offset, but to fix the weird animation that happens when using the interactive pop gesture in iOS11.0-11.1.

Now that it's centralized, we can at least elaborate the comment in one place, as the current one's pretty opaque.

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m (hunk 1 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: It's related to this fix, in that it's about contentInset, but I'm not sure if adding this code to the conversation details is necessary or will fix anything. 

I guess what I'm asking for is:

- have a full page of content 
- scroll all the way down.
- verify the home button isn't overlaying any content.

And to check that this holds for iOS10, iOS11.0-1, and iOS11.2

Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: > verify the home button isn't overlaying any content.

I think that's now true.  The problem is that now we're adding "double content inset" to account for the home view button, at least in some cases.

File: SignalMessaging/categories/UIView+OWS.m (hunk 2 above)
Comment (charlesmchen-signal) [reply SignalMessaging/categories/UIView+OWS.m]: Improved the comment.

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: Verified that content insets are appropriate on iOS 9 and iOS 11 and iPhone X.

Lots of screenshots here:
[ScrollViewInsetsViPhoneXScreenshots.zip](https://github.com/signalapp/Signal-iOS/files/1908582/ScrollViewInsetsViPhoneXScreenshots.zip)

**Pull request merged**